search/errors.py

```diff
@@ -0,0 +1,41 @@
+from flask import Blueprint, Response, abort, request
+
+from lxml import etree
+from werkzeug.exceptions import HTTPException
+
+NAMESPACE_MAP = {None: 'urn:ietf:rfc:7807'}
+
+
+def http_error_handler(error: HTTPException) -> Response:
+    root = etree.Element('problem', nsmap=NAMESPACE_MAP)
+    root.attrib['{http://www.w3.org/XML/1998/namespace}lang'] = "en"
+
+    status = etree.SubElement(root, 'status')
```

Comment:
that looks like a awkward API that we are forced to call for the creation of the document; @gnott do you have suggestions for improving the ease with which a small XML document can be created with `lxml`?

(I'd still avoid harcoding a XML string if it has to have dynamic values in it like these fields)

Reply:
e.g. the [`lxml.builder` module](https://lxml.de/api/lxml.builder.ElementMaker-class.html)? Sample for HTML, but it's similar:
```
page = (
    E.html(
        E.head(
            E.title("This is a sample document")
        ),
        E.body(
            E.h1("Hello!", CLASS("title")),
            E.p("This is a paragraph with ", B("bold"), " text in it!"),
            E.p("This is another paragraph, with a ",
                A("link", href="http://www.python.org"), "."),
            E.p("Here are some reserved characters: <spam&egg>."),
            ET.XML("<p>And finally, here is an embedded XHTML fragment.</p>"),
        )
    )
)
```

Reply:
@giorgiosironi Are you referring to the namespace on line 11?

Reply:
To the fact that each element needs to be created as a `SubElement` object and then its `text` field set separately, can be annoying to set up a (longer) document this way?

Reply:
updated.